diff --git a/Makefile b/Makefile
index 66f6681..160d145 100644
--- a/Makefile
+++ b/Makefile
@@ -1,10 +1,10 @@
 # Makefile hex2bin/mot2bin
 
-CPFLAGS = -std=c99 -O2 -Wall -pedantic
+CPFLAGS = -std=c99 -Wall -pedantic $(CFLAGS)
 
 # Compile
 %.o : %.c
-	gcc -c $(CPFLAGS) $< -o $@
+	$(CC) -c $(CPFLAGS) $< -o $@
 
 #WIN64_GCC = x86_64-w64-mingw32-gcc
 WINDOWS = i686-w64-mingw32
@@ -20,10 +20,10 @@ hex2bin.1: hex2bin.pod
 	pod2man hex2bin.pod > hex2bin.1
 
 hex2bin: hex2bin.o common.o libcrc.o binary.o
-	gcc -O2 -Wall -o hex2bin hex2bin.o common.o libcrc.o binary.o
+	$(CC) -Wall $(CFLAGS) -o hex2bin hex2bin.o common.o libcrc.o binary.o
 
 mot2bin: mot2bin.o common.o libcrc.o binary.o
-	gcc -O2 -Wall -o mot2bin mot2bin.o common.o libcrc.o binary.o
+	$(CC) -Wall $(CFLAGS) -o mot2bin mot2bin.o common.o libcrc.o binary.o
 
 windows:
 #	$(WIN32_GCC) $(CPFLAGS) -o Win32/hex2bin.exe hex2bin.c common.c libcrc.c binary.c
@@ -34,10 +34,12 @@ windows:
 	$(WIN_STRIP) Win64/mot2bin.exe
 
 install:
-	strip hex2bin
-	strip mot2bin
-	cp hex2bin mot2bin $(INSTALL_DIR)/bin
-	cp hex2bin.1 $(MAN_DIR)
+#	strip hex2bin
+#	strip mot2bin
+	install -m 755 -d $(INSTALL_DIR)/bin
+	install -m 755 -d $(MAN_DIR)
+	install -m 755 hex2bin mot2bin $(INSTALL_DIR)/bin/
+	install -m 644 hex2bin.1 $(MAN_DIR)/
 
 clean:
 	rm core *.o hex2bin mot2bin
